### PR TITLE
Get data us

### PR DESCRIPTION
**Pull request by jhellewell14**

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Generated by roxygen2: do not edit by hand
 
 export("%>%")
+export(get_US_regional_cases)
```
Comment (seabbs) [inline NAMESPACE:4]: no caps for the function name please

File: R/get_US_regional_cases.R [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#' Fetch daily COVID cases by state for United States of America
+#'
+#' @return
+#' @export
+#' @importFrom memoise cache_filesystem memoise
+#' @importFrom dplyr filter select mutate arrange group_by n lag
+#' @importFrom tidyr gather
+#' @importFrom lubridate mdy
+#' @examples
+#'
```
Comment (seabbs) [inline R/get_US_regional_cases.R:18]: Example can have the code or an example of use

File: R/get_US_regional_cases.R [hunk 3]
```diff
@@ -0,0 +1,49 @@
+
+#' Fetch daily COVID cases by state for United States of America
+#'
+#' @return
```
Comment (seabbs) [inline R/get_US_regional_cases.R]: return shouldnt be empty

File: R/get_US_regional_cases.R [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  path <- "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Confirmed.csv"
+
+  ch <- memoise::cache_filesystem(".cache")
+
+
+  mem_read <- memoise::memoise(readr::read_csv, cache = ch)
+
+  cases <- mem_read(path)
+
+  cases <- cases %>% dplyr::filter(`Province/State` %in% state.name) %>%
```
Comment (seabbs) [inline R/get_US_regional_cases.R]: new line per operation please

File: R/get_US_regional_cases.R [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    dplyr::select(-`Country/Region`,-Lat,-Long, region = `Province/State`) %>%
```
Comment (seabbs) [inline R/get_US_regional_cases.R]: spaces

File: R/get_US_regional_cases.R [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    dplyr::group_by(region) %>%
+    dplyr::mutate(
+      index = 1:dplyr::n(),
+      cases = total_cases - ifelse(index == 1, 0, dplyr::lag(total_cases))) %>%
+    dplyr::ungroup() %>%
+    dplyr::select(-index, -total_cases) %>%
+    ## Adjust negative cases by setting to 0
+    dplyr::mutate(cases = ifelse(cases < 0 , 0, cases)) %>%
+    dplyr::filter(region != "Grand Princess", region != "Diamond Princess") %>%
+    dplyr::full_join(data.frame(region = state.name, region_code = state.abb), by = "region")
```
Comment (seabbs) [inline R/get_US_regional_cases.R:49]: what happens here if a state has no cases and we full join? A bunch of NAs? May need to fill with zeros

File: R/get_US_regional_cases.R [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  # Example plotting code
```
Comment (seabbs) [inline R/get_US_regional_cases.R]: please add as a \dontrun{} example (as for germany)

**Review by seabbs — CHANGES_REQUESTED** [review]
Looks good in general. A few things to update (being picky). 

Also could you add function to the following:

* README
* pkgdown.yml

and add attribution in the function description

File: R/get_US_regional_cases.R (hunk 6 above)
Comment (jhellewell14) [reply R/get_US_regional_cases.R:49]: They already all exist so it's fine

**Review by seabbs — APPROVED** [review]
An exceptionally pleasing pull request

**Comment by seabbs** [comment]
build failure is due to an old namespace - will clean on master

**Pull request merged**